sdk/agrifood/azure-agrifood-farming/samples/async/sample_attachments_async.py

```diff
@@ -0,0 +1,202 @@
+            await client.attachments.get(
+                farmer_id=farmer_id,
+                attachment_id=attachment_on_farm_id
+            )
+            print("Attachment already exists. Not updating file.")
+
+        except ResourceNotFoundError:
+            print("Attachment doesn't exist")
+            print("Creating attachment...", end=" ", flush=True)
+
+            # Open file with buffering set to 0, to get a IO object.
+            file_to_attach_on_farm = open(
+                attachment_on_farm_file_path,
+                "rb",
+                buffering=0)
+
+            await client.attachments.create_or_update(
+                farmer_id=farmer_id,
+                attachment_id=attachment_on_farm_id,
+                resource_id= farm_id,
+                resource_type="Farm",
+                file=file_to_attach_on_farm)
+
+            print("Done!")
+        
+        print("Proceeding to download all attachments on the farmer. " +
+            "Press enter to continue...")
+        input()
+
+
+        print("Getting a list of all attachments " +
+            f"on the farmer with id {farmer_id}...", end=" ", flush=True)
+        farmer_attachments = client.attachments.list_by_farmer_id(
+            farmer_id=farmer_id,
+        )
+        print("Done!")
+
+        # Using a semaphore to limit the number of concurrent downloads.
+        semaphore =  asyncio.Semaphore(2)
+        
+
+        print("Downloading attachments with a maximum concurrency "+
+            "of two downloads at a time...")
+
+        # Setting up a async function (a coroutine) to download each attachment
+        async def download(attachment, semaphore):
+            async with semaphore:
+                downloaded_attachment = await client.attachments.download(
+                    farmer_id=farmer_id,
+                    attachment_id=attachment_on_farmer_id
+                )
+                out_path = \
+                    "../data/attachments/" + \
+                    f"{attachment.resource_type}/{attachment.resource_id}" + \
+                    f"/{attachment.id}/{attachment.original_file_name}"
+
+                # Make sure the dirs to the output path exists
+                Path(out_path).parent.mkdir(parents=True, exist_ok=True)
+
+                print(f"Saving attachment id {attachment.id} to {out_path}")
+                with open(
+                        out_path,
+                        'wb'
+                        ) as out_file:
+                    async for bits in downloaded_attachment:
+                            out_file.write(bits)
+
+        await asyncio.gather(
+            *[download(attachment, semaphore) async for attachment in farmer_attachments]
+        )
+            
+
+        print("All files downloaded into directory 'data/attachments/'")
+
+
+async def main():
+    # Get farmbeats endpoint
+    try:
+        farmbeats_endpoint = os.environ['FARMBEATS_ENDPOINT']
+    except KeyError:
```

Comment:
there's no need to do a try except, just directly access the env vars. That will throw the necessary KeyError, without adding code